### PR TITLE
Theme Next: unified color palette

### PR DESCRIPTION
**Pull request by origami-z**
Apply new unified color palette (foundation) when using `UNSTABLE_SaltProviderNext`, with remapped palette and characteristics layer as well. This does not introduce new Button design (w/ top/bottom border states) but still using existing background change. This however does introduce _mode agnostic_ Buttons, a.k.a. Button in both light and dark mode will share the same color / state.

Note ag grid theme doesn't work in theme next, mostly due to [default color applied](https://github.com/jpmorganchase/salt-ds/blob/98d1a03fbdeb493ec235c41e6009d51f2a614675/packages/ag-grid-theme/css/salt-ag-theme.scss#L320-L328). We don't care about them for now, and will be easily fixed when v31 ag grid support is released.

- [x] Final color swatch r8 revision
- [x] Button color mapping check

Added border to button to accommodate new design. This means some components / test are touched
- Button, to keep visual regression the same, updated padding to deduct border
- Segmented button separator changed implementation to have full height stroke
- Input, Multiline Input, Pill Input, updated padding to deduct border

Replace #3032

### CODE REVIEW

File: packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss [hunk 1]
```diff
@@ -11,6 +11,11 @@
     border: none;
   }
 
+  .ag-root-wrapper,
```
Comment (origami-z) [inline packages/ag-grid-theme/css/_salt-ag-theme-mixin.scss:14]: These 2 shouldn't be needed. Need to work out why these doesn't work.

File: packages/ag-grid-theme/css/_salt-ag-theme-default-params.scss [hunk 2]
```diff
@@ -3,7 +3,7 @@
 $salt-ag-theme-default-params: (
   row-hover-color: var(--agGrid-row-color-hover),
   row-border-color: var(--agGrid-row-borderColor),
-  header-background-color: var(--salt-palette-neutral-primary-background),
+  header-background-color: var(--salt-container-primary-background),
```
Comment (origami-z) [inline packages/ag-grid-theme/css/_salt-ag-theme-default-params.scss:6]: Will be updated in #3363

File: packages/theme/css/characteristics/navigable-next.css [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.salt-theme.salt-theme-next {
+  --salt-navigable-indicator-hover: var(--salt-palette-neutral);
+  --salt-navigable-indicator-active: var(--salt-palette-accent);
+
+  /* --salt-navigable-background-hover: var(--salt-palette-navigate-background-hover); */
```
Comment (joshwooding) [inline packages/theme/css/characteristics/navigable-next.css:5]: Is this meant to be commented out?

**Review by joshwooding — APPROVED** [review]

File: packages/theme/css/characteristics/navigable-next.css (hunk 3 above)
Comment (origami-z) [reply packages/theme/css/characteristics/navigable-next.css:5]: This is a deprecated color apparently

File: packages/theme/css/characteristics/target-next.css [hunk 4]
```diff
@@ -0,0 +1,4 @@
+.salt-theme.salt-theme-next {
+  --salt-target-background-hover: var(--salt-palette-accent-default);
```
Comment (libertymayc) [inline packages/theme/css/characteristics/target-next.css:2]:
```suggestion
  --salt-target-background-hover: var(--salt-palette-accent);
```

Comment (libertymayc) [reply packages/theme/css/characteristics/target-next.css:2]: picked this up running validation on https://github.com/jpmorganchase/salt-ds/pull/3143

**Review by lilyvc — APPROVED** [review]

**Pull request merged**